### PR TITLE
TabLayoutMediaLibrary

### PR DESCRIPTION
**Pull request by Rai-gou**

### CODE REVIEW

**Comment by IvanGusevE** [comment]
**_Критическая_**

На экране медиатека для всего текста нужно задать font YS/Medium:
image

**Comment by IvanGusevE** [comment]
**_Критическая_**
Отступ между табами и заглушкой должен быть 106.dp:

image

File: app/src/main/java/com/example/myplaylist/library/ui/MediaLibraryActivityViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private val libraryLoadingLiveData = MutableLiveData(ScreenState())
+    val getLibraryLoadingLiveData: LiveData<ScreenState> get() = libraryLoadingLiveData
+
+    fun setupTabLayoutMediator(tabLayout: TabLayout, viewPager: ViewPager2) {
+        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
+            when (position) {
+                0 -> tab.text = context.getString(R.string.favorite_tracks)
+                1 -> tab.text = context.getString(R.string.playlists)
+            }
+        }.attach()
```
Comment (IvanGusevE) [inline app/src/main/java/com/example/myplaylist/library/ui/MediaLibraryActivityViewModel.kt]: **_Критическая_**
Инициализацию **TabLayoutMediator** лучше делать в **активити**. ViewModel предназначена в первую очередь для того, чтобы проксировать события между presentation и domain слоем. Также ViewModel живет дольше активити (переживает смену конфигурации и тд.). Делегируя инициализацию вьюшек во вью модель, ты рискуешь получить утечку памяти (активити может пересоздаться, а viewmodel будет продолжать держать ссылку на вьюшку, которая была привязана к разметке прошлой активити).
Также в методе onDestroy нужно вызвать detach(), чтобы отвязать TabLayoutMediator:

```
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mediator = TabLayoutMediator(tabLayout, viewPager) { tab, position ->
            ...
        }.attach()
}

override fun onDestroy() {
        mediator.detach()
        super.onDestroy()
}

```

File: app/src/main/java/com/example/myplaylist/library/ui/PlaylistFragment.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import com.example.myplaylist.databinding.FragmentPlaylistsBinding
+import org.koin.android.ext.android.inject
+
+class PlaylistFragment : Fragment() {
+
+    private lateinit var binding: FragmentPlaylistsBinding
```
Comment (IvanGusevE) [inline app/src/main/java/com/example/myplaylist/library/ui/PlaylistFragment.kt]: **_Критическая_**

Представление фрагмента (его визуальный интерфейс или View) имеет отдельный жизненный цикл. Поэтому необходимо удалить все ссылки на экземпляр класса привязки в onDestroyView().
Можно посмотреть пример реализации по этой [ссылке]( https://developer.android.com/topic/libraries/view-binding).

image

File: app/src/main/java/com/example/myplaylist/library/ui/MediaLibraryActivityViewModel.kt [hunk 3]
```diff
@@ -4,23 +4,11 @@ import android.content.Context
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import androidx.viewpager2.widget.ViewPager2
-import com.example.myplaylist.R
 import com.example.myplaylist.search.data.ScreenState
-import com.google.android.material.tabs.TabLayout
-import com.google.android.material.tabs.TabLayoutMediator
 
 class MediaLibraryActivityViewModel(private val context: Context): ViewModel() {
```
Comment (IvanGusevE) [inline app/src/main/java/com/example/myplaylist/library/ui/MediaLibraryActivityViewModel.kt:9]: Контекст больше не используется во viewmodel (можно убрать).

File: app/src/main/java/com/example/myplaylist/library/ui/PlaylistFragment.kt [hunk 4]
```diff
@@ -5,20 +5,24 @@ import android.view.LayoutInflater
...
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.ViewModelProvider
 import com.example.myplaylist.databinding.FragmentPlaylistsBinding
 import org.koin.android.ext.android.inject
 
 class PlaylistFragment : Fragment() {
 
-    private lateinit var binding: FragmentPlaylistsBinding
+    private var _binding: FragmentPlaylistsBinding? = null
+    private val binding get() = _binding!!
```
Comment (IvanGusevE) [inline app/src/main/java/com/example/myplaylist/library/ui/PlaylistFragment.kt:14]: Супер!
